index.mjs

```diff
@@ -146,23 +143,34 @@ ParallelHook.prototype[$map] = function(f){
   return ParallelHook(map(f)(this.hook));
 }
 
+const crash = e => Future(() => { throw e });
+
 ParallelHook.prototype[$ap] = function(mf){
   return ParallelHook(Hook(c => {
-    let f, x, b = false;
-    const mirror = cache(never);
-    const rf = mf.hook.run(_f => {
-      f = _f;
-      if (b) return map(x => mirror.resolve(x) || x)(c(f(x)));
-      b = true;
-      return mirror;
-    });
-    const rx = this.hook.run(_x => {
-      x = _x;
-      if (b) return map(x => mirror.resolve(x) || x)(c(f(x)));
-      b = true;
-      return mirror;
-    });
-    return map(head)(both(rf, rx));
```

Comment:
Usage of `both`, combined with the externally-triggered resolution (previously through `mirror.resolve`, now through the capture `res` function called from `consume`), was the cause of the issue.